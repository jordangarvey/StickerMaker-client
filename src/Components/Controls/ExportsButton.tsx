import React, { FC } from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

import * as C from "../../Global/Colours";


/**
 * Interface for the Template Item props
 */
interface IExportsButtonProps {
	/** Callback to handle the image copying */
	onCopy(): void;
}

/**
 * Component to render a Category button
 */
const ExportsButton: FC<IExportsButtonProps> = (props) => {
	return (
		<TouchableOpacity
			onPress={props.onCopy}
			style={styles.exportsButton}
		>
			<Text style={styles.buttonText}>Copy to clipboard</Text>
		</TouchableOpacity>
	);
};

const styles = StyleSheet.create({
	buttonText: {
		color: C.secondary,
		fontSize: 16,
		fontWeight: "bold"
	},
	exportsButton: {
		alignItems: "center",
		backgroundColor: C.primary,
		borderBottomLeftRadius: 30,
		borderBottomRightRadius: 30,
		borderTopLeftRadius: 6,
		borderTopRightRadius: 6,
		bottom: 20,
		flexDirection: "row",
		height: 70,
		justifyContent: "center",
		left: "10%",
		marginTop: 40,
		padding: 5,
		width: "90%"
	}
});

export default ExportsButton;
