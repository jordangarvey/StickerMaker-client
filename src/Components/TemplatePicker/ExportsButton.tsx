import React, { SFC } from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

import * as C from "../../Global/Colours";


/**
 * Component to render a Category button
 */
const ExportsButton: SFC = () => (
	<TouchableOpacity
		style={styles.exportsButton}
	>
		<Text style={styles.buttonText}>Export to camera roll</Text>
	</TouchableOpacity>
);

/**
 * Styles for the Category button
 */
const styles = StyleSheet.create({
	buttonText: {
		color: C.secondary,
		fontSize: 16,
		fontWeight: "bold"
	},
	exportsButton: {
		alignItems: "center",
		backgroundColor: C.primary,
		borderBottomLeftRadius: 30,
		borderBottomRightRadius: 30,
		borderTopLeftRadius: 6,
		borderTopRightRadius: 6,
		height: 70,
		flexDirection: "row",
		justifyContent: "center",
		marginHorizontal: 20,
		padding: 5,
		width: "100%"
	}
});

export default ExportsButton;
