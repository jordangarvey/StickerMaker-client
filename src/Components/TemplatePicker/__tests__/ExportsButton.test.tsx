import React from "react";

import renderer from "react-test-renderer";

import ExportsButton from "../ExportsButton";


describe("The Exports Button componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		const component = renderer.create(
			<ExportsButton/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
