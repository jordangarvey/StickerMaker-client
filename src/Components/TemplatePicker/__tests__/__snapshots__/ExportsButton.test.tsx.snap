// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Exports Button componentâ€™s snapshot renders as expected 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#18A0FB",
      "borderBottomLeftRadius": 30,
      "borderBottomRightRadius": 30,
      "borderTopLeftRadius": 6,
      "borderTopRightRadius": 6,
      "flexDirection": "row",
      "height": 70,
      "justifyContent": "center",
      "marginHorizontal": 20,
      "opacity": 1,
      "padding": 5,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Export to camera roll
  </Text>
</View>
`;
